scripts/xtensa-build-zephyr.sh

```diff
@@ -430,7 +431,9 @@ see https://docs.zephyrproject.org/latest/getting_started/index.html"
 		mkdir -p "${WEST_TOP}"/modules/audio
 		ln -s "$SOF_TOP" "${WEST_TOP}"/modules/audio/sof
 	}
-
+        if [[ $IPC4_BUILD == true ]]; then
+	  echo "CONFIG_IPC_MAJOR_4=y" >> ${SOF_TOP}/zephyrproject/zephyr/samples/subsys/audio/sof/boards/intel_adsp_cavs25.conf
```

Comment:
This only builds for one target, can we make it target generic. @marc-hb any suggestions how how this would be done here ?

Reply:
@marc-hb Except for adding CONFIG_IPC_MAJOR_4=y here,
according to @RanderWang,CONFIG_RIMAGE_SIGNING_SCHEMA need to be modified from "tgl-h" to "tgl-cavs" too
```
CONFIG_RIMAGE_SIGNING_SCHEMA="tgl-cavs"
```

Reply:
But still encountered  DSP topology loading failure with this fw and innersource topic sof cavs dev kernel.
Will post an issue later

Reply:
Sorry I missed this.

We can't do it like this because this pollutes the `git diff` and it will also keep growing the .conf file in people's workspaces.

Zephyr has a number of Kconfig options: 
https://docs.zephyrproject.org/2.7.0/guides/build/kconfig/setting.html#the-initial-configuration

Reply:
The document doesn't mention application level kconfig overriding, but according to
```
5. Otherwise, if board revisions are used and boards/<BOARD>_<revision>.conf exists in the application directory, the result of merging it with prj.conf and boards/<BOARD>.conf is used.
```
we can workaround it by caching a `<BOARD>_<revision>.conf `, and copy it to zephyr tree when build SOF+Zephyr+IPC4. @XiaoyunWu6666  can you do this?